### PR TITLE
Bugfix/optimize validation

### PR DESCRIPTION
**Pull request by DawidMakar-acaisoft**
The changeset consists of:

- rewriting `validate_values_in_df` from applying a complex function for each row to performing all the checks using vectorized operations, which gives a huge speedup
- simplifying value parser conditions
- fixing tests

### CODE REVIEW

File: opcua_tools/nodeset_generator.py [hunk 1]
```diff
@@ -336,7 +335,7 @@ def create_nodeset2_file(
         nodes=nodes, references=references, namespace_index=serialize_namespace
     )
     namespaces_in_use.sort()
-    original_nodes = nodes.copy()
+    data_type_nodes = nodes[nodes["NodeClass"] == "UADataType"].copy()
     nodes = nodes[nodes["ns"].map(lambda x: x in (namespaces_in_use))].copy()
```
Comment (olavpe) [inline opcua_tools/nodeset_generator.py:339]: Nice! sending and casting just want is needed is great:))

File: opcua_tools/value_parser.py [hunk 2]
```diff
@@ -99,61 +99,61 @@ def parse_singular_value(val, tagtype):
         else:
             stripped = None
         if tagtype == "SByte":
-            if stripped is not None and stripped != "":
```
Comment (olavpe) [inline opcua_tools/value_parser.py:102]: Have you checked these changes all the way up to model-builder tests using the packages locally?

**Review by olavpe — APPROVED** [review]
Looks good, just curious if the "stripped" changes have been tested all the way up to the model-builder?

File: opcua_tools/value_parser.py (hunk 2 above)
Comment (DawidMakar-acaisoft) [reply opcua_tools/value_parser.py:102]: I've just double-checked it - works well in all the places (tools/mapper/MB).

**Pull request merged**